### PR TITLE
Edited `shouldNodeComponentUpdate` documentation

### PR DESCRIPTION
**Pull request by dekdekbaloo**
#### Is this adding or improving a _feature_ or fixing a _bug_?
It's a minor document editing.

#### What's the new behavior?
The actual place where `shouldNodeComponentUpdate` function hooks in was wrongly documented as `static shouldNodeComponentUpdate(props, nextProps)` in [Custom nodes](docs/reference/slate-react/custom-nodes.md) documentation.
I've edited and linked the documentation to [Plugins](docs/refererence/slate-react/plugins.md) page since that's where the function is really called.

#### How does this change work?
As I looked through the implementation in [node.js](https://github.com/ianstormtaylor/slate/blob/3339d088e14ad0bf04a674c14834179403ae369f/packages/slate-react/src/components/node.js#L66), the function was found from the plugin's `stack` instead of the `component`'s static property and that's why having `static shouldNodeComponentUpdate(props, nextProps)` in your own component won't work.

Fixes: #

Reviewers: @

### CODE REVIEW

**Comment by ianstormtaylor** [comment]
Hey @dekdekbaloo this looks great, thanks for doing this! Can you fix the linter issues that are causing CI to fail?

File: docs/reference/slate-react/plugins.md [hunk 1]
```diff
@@ -32,6 +32,7 @@ export default function MySlatePlugin(options) {
   onKeyUp: Function,
   onPaste: Function,
   onSelect: Function
```
Comment (davidchang) [inline docs/reference/slate-react/plugins.md]: nit: can you add a trailing comma?

File: docs/reference/slate-react/plugins.md [hunk 2]
```diff
@@ -112,6 +113,9 @@ _Note: This is **not** Slate's internal selection representation (although it mi
 ```js
 {
   onChange: Function
+  renderEditor: Function
```
Comment (davidchang) [inline docs/reference/slate-react/plugins.md]: can you add relevant commas here?

File: docs/reference/slate-react/plugins.md [hunk 3]
```diff
@@ -132,3 +136,23 @@ The `renderEditor` property allows you to define higher-order-component-like beh
 `Object`
 
 The `schema` property allows you to define a set of rules that will be added to the editor's schema. The rules from each of the schemas returned by the plugins are collected into a single schema for the editor.
+
+### `shouldNodeComponentUpdate`
+
+`Function shouldNodeComponentUpdate(props: Object, nextProps: Object) => boolean || null`
+
+To everride slate's default `shouldComponentUpdate` behavior, add this property in and define any custom logic. For example:
```
Comment (davidchang) [inline docs/reference/slate-react/plugins.md]: *override

**Comment by dekdekbaloo** [comment]
@ianstormtaylor @davidchang Hey guys, I'm so sorry, I've been very busy and I didn't have time to fix the PR. Will start working on this asap. :)

**Comment by dekdekbaloo** [comment]
@ianstormtaylor Hello, I'm not quite sure which part leads to the linting errors because I did only just documentation change. Could you please help me looking through this or at least give me some guidance?

**Comment by ianstormtaylor** [comment]
As of https://github.com/ianstormtaylor/slate/pull/3093 (which was just merged), I believe this issue is no longer applicable, because a lot has changed. I'm going through and closing out any potential issues that are not out of date with the overhaul. Thanks for understanding.

**Pull request closed without merging**